### PR TITLE
Cleanup: git >= 2.32 does not support gitignore as symlink.

### PR DESCRIPTION
**Pull request by andreaspeters**
Nothing important, only a cleanup.

Git version 2.32.0 does not support gitignore as symlink ([https://github.com/git/git/blob/master/Documentation/RelNotes/2.32.0.txt](https://github.com/git/git/blob/master/Documentation/RelNotes/2.32.0.txt)). I updated setup-dev.sh to copy gitignore into the mesos root path. I also add some more filenames into gitignore. All of them was dynamically created by bootstrap.sh, setup-dev.sh and/or autoconf/make.

### CODE REVIEW

File: support/setup-dev.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ __EOF__
 fi
 
 if test ! -e .gitignore; then
```
Comment (cf-natali) [inline support/setup-dev.sh:29]: Maybe add a comment explaining why a copy is necessary, pointing to https://github.com/git/git/blob/master/Documentation/RelNotes/2.32.0.txt ?

File: support/gitignore [hunk 2]
```diff
@@ -39,3 +39,27 @@ CPPLINT.cfg
...
+aclocal.m4
+ar-lib
+autom4te.cache/
+build/
+centos7/
+compile
+config.guess
+config.sub
+configure
+*~
```
Comment (cf-natali) [inline support/gitignore:60]: This `*~` will catch many files, are there many such files which can't be listed individually?

Comment (andreaspeters) [reply support/gitignore]: I changed my mind, I think that does not have to be there. With that I want catch tmp files from my editor. But maybe that's to much. :sweat_smile:

File: support/setup-dev.sh (hunk 1 above)
Comment (andreaspeters) [reply support/setup-dev.sh:29]: :+1:

File: support/packaging/centos/build-docker-centos.sh [hunk 3]
```diff
@@ -24,7 +24,7 @@ DOCKER_CONTEXT_DIR="${SOURCE_DIR}/centos${CENTOS_DISTRO}/rpmbuild/RPMS/x86_64"
 
 cat <<EOF > "${DOCKER_CONTEXT_DIR}/Dockerfile"
 FROM centos:${CENTOS_DISTRO}
-ADD mesos-?.?.?-*.rpm /
+ADD mesos-*.rpm /
```
Comment (cf-natali) [inline support/packaging/centos/build-docker-centos.sh:27]: This change seems unrelated?

File: support/packaging/centos/mesos.spec [hunk 4]
```diff
@@ -1,9 +1,6 @@
-%global MESOS_VERSION
-%global MESOS_RELEASE
-
 Name:          mesos
-Version:       %{MESOS_VERSION}
-Release:       %{MESOS_RELEASE}%{?dist}
+Version:       1.11.1
```
Comment (cf-natali) [inline support/packaging/centos/mesos.spec:2]: This file too?

Comment (andreaspeters) [reply support/packaging/centos/mesos.spec:2]: Oh  no... Good that u look at it. :-)  This and the upper one does not have to be there. I will cleanup. Sorry! :-(

File: support/gitignore [hunk 5]
```diff
@@ -39,3 +39,26 @@ CPPLINT.cfg
 .deps/
 .libs/
 *-stamp
+
+.clang-format
```
Comment (cf-natali) [inline support/gitignore:43]: Several of those files already appear above, e.g. `clang-format`, `Makefile.in`, etc - could you make sure there are no duplicates?

Also, it would be nice to add them under the relevant sections.

**Comment by cf-natali** [comment]
Hey, I suggest you only keep the `gitignore` copy in this MR, and create a new MR adding entries to `gitignore`.

**Comment by cf-natali** [comment]
No worries, hope you had a nice break!
